### ISSUE TITLE
2.15.6 crash on Rails deploy at rollbar:sourcemap stage

### ISSUE THREAD

**Issue by DMA57361** [rollbar/rollbar-gem#713]
TLDR: Moving from 2.15.5 to 2.15.6 has added a new `rollbar:sourcemap` stage to my capistrano deploys, which crashes unless I add a `rollbar_sourcemaps_minified_url_base` config value (even though I have no meaningful value to set on most of our apps).

Below is relevant bit from the `bundle exec cap production deploy --trace` (partly anonymised):

```
** Invoke deploy:assets:backup_manifest (first_time)
** Execute deploy:assets:backup_manifest
00:14 deploy:assets:backup_manifest
      01 mkdir -p /var/www/projectname/releases/20180418084937/assets_manifest_backup
    ✔ 01 deploy@server.hostname.com 0.054s
      02 cp /var/www/projectname/releases/20180418084937/public/assets/.sprockets-manifest-ba1494b464b023103745788f3cd11928.json /var/www/projectname/releases/20180418084937/assets_manifest_backup
    ✔ 02 deploy@server.hostname.com 0.056s
** Invoke rollbar:sourcemap (first_time)
** Execute rollbar:sourcemap
00:14 rollbar:sourcemap
      Uploading source maps from
#<Thread:0x00000001055bf788@/Users/username/.rvm/gems/ruby-2.5.1@projectname/gems/sshkit-1.16.0/lib/sshkit/runners/parallel.rb:10 run> terminated with exception (report_on_exception is true):
Traceback (most recent call last):
	1: from /Users/username/.rvm/gems/ruby-2.5.1@projectname/gems/sshkit-1.16.0/lib/sshkit/runners/parallel.rb:11:in `block (2 levels) in execute'
/Users/username/.rvm/gems/ruby-2.5.1@projectname/gems/sshkit-1.16.0/lib/sshkit/runners/parallel.rb:15:in `rescue in block (2 levels) in execute': Exception while executing as deploy@server.hostname.com: undefined local variable or method `rollbar_sourcemaps_minified_url_base' for #<SSHKit::Backend::Netssh:0x00000001055bf620> (SSHKit::Runner::ExecuteError)
cap aborted!
SSHKit::Runner::ExecuteError: Exception while executing as deploy@server.hostname.com: undefined local variable or method `rollbar_sourcemaps_minified_url_base' for #<SSHKit::Backend::Netssh:0x00000001055bf620>
/Users/username/.rvm/gems/ruby-2.5.1@projectname/gems/sshkit-1.16.0/lib/sshkit/runners/parallel.rb:15:in `rescue in block (2 levels) in execute'
/Users/username/.rvm/gems/ruby-2.5.1@projectname/gems/sshkit-1.16.0/lib/sshkit/runners/parallel.rb:11:in `block (2 levels) in execute'

Caused by:
NameError: undefined local variable or method `rollbar_sourcemaps_minified_url_base' for #<SSHKit::Backend::Netssh:0x00000001055bf620>
/Users/username/.rvm/gems/ruby-2.5.1@projectname/gems/rollbar-2.15.6/lib/rollbar/tasks/rollbar.cap:39:in `block (3 levels) in <top (required)>'
/Users/username/.rvm/gems/ruby-2.5.1@projectname/gems/sshkit-1.16.0/lib/sshkit/backends/abstract.rb:29:in `instance_exec'
/Users/username/.rvm/gems/ruby-2.5.1@projectname/gems/sshkit-1.16.0/lib/sshkit/backends/abstract.rb:29:in `run'
/Users/username/.rvm/gems/ruby-2.5.1@projectname/gems/sshkit-1.16.0/lib/sshkit/runners/parallel.rb:12:in `block (2 levels) in execute'
Tasks: TOP => rollbar:sourcemap
The deploy has failed with an error: Exception while executing as deploy@server.hostname.com: undefined local variable or method `rollbar_sourcemaps_minified_url_base' for #<SSHKit::Backend::Netssh:0x00000001055bf620>
** Invoke deploy:failed (first_time)
** Execute deploy:failed
```

Having had a glance at the top of the backtrace - https://github.com/rollbar/rollbar-gem/blob/master/lib/rollbar/tasks/rollbar.cap#L39 - it should probably just be a string, without the interpolation.

I assume setting a `rollbar_sourcemaps_minified_url_base` value will fix the problem, however this application (any several others of ours) has no sourcemaps and is also firewalled to be internal to our company. So I don't have a useful value to assign here.

I tested this with 3 of our applications and `bundle update rollbar` then `bundle exec cap production deploy` is enough to cause the exception.

I'm rolling back to 2.15.5 for now, but thought I'd let you know.

**Comment by noelrappin** [comment]
Me too

**Comment by yoonwaiyan** [comment]
Same here. Looking at the codebase seems like putting a default value to the `fetch` will do the trick: `url_base = fetch(:rollbar_sourcemaps_minified_url_base, nil)`.

**Comment by CodingAnarchy** [comment]
This just caused me some headaches too.

**Comment by rivkahstandig3636** [comment]
We will take a look into this, sorry for the delay!

**Comment by tisba** [comment]
Has this issue been resolved by v2.16.0? There are no updates to the CHANGELOG, only the [release page](https://github.com/rollbar/rollbar-gem/releases/tag/v2.16.0) has some info but none specific to this issue here.

**Comment by rdunlop** [comment]
no, this is issue is not resolved. We are still hitting it on 2.16.0 @tisba

**Comment by neohunter** [comment]
So apparently this is fixed by adding:

```
set :rollbar_sourcemaps_minified_url_base, "https://www.your-host.com"
```

to `config/deploy/production.rb`

However after doing that, the error not appears but this step on deploy:


```
01:16 rollbar:sourcemap
      Uploading source maps
``` have been running for abour 25 minutse without progress...

How to disable it?

**Comment by westonplatter** [comment]
I forked the rollbar repo and commented out the source maps upload feature to remove issue. https://github.com/westonplatter/rollbar-gem

**Comment by tomdracz** [comment]
Just opened #742 that should fix it. Still makes it dependant on capistrano rails assets task but at least it doesn't blow up now if the value of :rollbar_sourcemaps_minified_url_base is not provided or falsey

**Comment by dylancopeland** [comment]
This bug seems to still be present in 2.16.2 despite the [release notes](https://github.com/rollbar/rollbar-gem/releases/tag/v2.16.1) for 2.16.1 saying it was fixed. The last working version for our application is 2.15.5.

Am I missing something?

**Comment by jondeandres** [comment]
@dylancopeland I've been taking a look at this and https://github.com/rollbar/rollbar-gem/issues/725, which are caused by https://github.com/rollbar/rollbar-gem/pull/678. That PR was reverted on https://github.com/rollbar/rollbar-gem/pull/747 and there is no more references to sourcemaps in the capistrano task.

Maybe you didn't really upgrade the rollbar gem so it's still trying to upload the compiled sourcemaps?
